cpp/src/parquet/page_index.h

```diff
@@ -43,6 +46,12 @@ class PARQUET_EXPORT ColumnIndex {
                                            uint32_t index_len,
                                            const ReaderProperties& properties);
 
+  static std::unique_ptr<ColumnIndex> Make(const ColumnDescriptor& descr,
+                                           const void* serialized_index,
+                                           uint32_t index_len,
+                                           const ReaderProperties& properties,
+                                           Decryptor* decryptor);
```

Comment:
Perhaps we can simply `decryptor` an optional argument (defaulting to `NULLPTR`) to avoid adding overloads :-)